src/orders-v2/order-refunds/order-refunds-view-model.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  AppAvatarFragment,
+  OrderDetailsFragment,
+  OrderGrantedRefundFragment,
+  OrderGrantedRefundStatusEnum,
+  StaffMemberAvatarFragment,
+  TransactionActionEnum,
+  TransactionEventFragment,
+  TransactionEventTypeEnum,
+} from "@dashboard/graphql";
+
+export type OrderRefundDisplay = {
+  id: string;
+  type: "standard" | "manual";
+  status: OrderGrantedRefundStatusEnum;
+  amount: {
+    amount: number;
+    currency: string;
+  };
+  reason: string | null;
+  createdAt: string;
+  user: {
+    email: string;
+    firstName: string;
+    lastName: string;
+  } | null;
+};
+
+export type OrderRefundState =
+  | "noTransactionsToRefund"
+  | "allTransactionsNonRefundable"
+  | "refundable";
+
+export abstract class OrderRefundsViewModel {
+  static getRefundState(transactions: OrderDetailsFragment["transactions"]): OrderRefundState {
+    if (transactions.length === 0) {
+      return "noTransactionsToRefund";
+    }
+
+    if (
+      transactions.every(transaction => !transaction.actions.includes(TransactionActionEnum.REFUND))
+    ) {
+      return "allTransactionsNonRefundable";
+    }
+
+    return "refundable";
+  }
+
+  static canEditRefund(refund: OrderRefundDisplay): boolean {
+    const isSuccessful = refund.status === OrderGrantedRefundStatusEnum.SUCCESS;
+    const isPending = refund.status === OrderGrantedRefundStatusEnum.PENDING;
+    const isManual = refund.type === "manual";
+
+    // Can only edit refunds that are not successful, not pending, and not manual
+    return !isSuccessful && !isPending && !isManual;
+  }
+
+  private static groupEventsByPspReference = (
+    events: TransactionEventFragment[],
+  ): Record<string, TransactionEventFragment[]> => {
+    return events?.reduce<Record<string, TransactionEventFragment[]>>((acc, event) => {
+      if (!acc[event.pspReference]) {
+        acc[event.pspReference] = [];
+      }
+
+      acc[event.pspReference].push(event);
+
+      return acc;
+    }, {});
+  };
+
```

Comment:
I know you copy-pasted it, but there is a bug (you can check my PR with refund reasons)

1. Create refund that fails
2. Create another refund that fails

Both of them have the same PSP (from transaction id? not sure). Eventually they are all merged and you will see only the last event.

I think we should see all of them

(non blocking, we can park it for later, details are visible in events list anyway)